galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/PropertyRoute.java

```diff
@@ -0,0 +1,196 @@
+        String responseBody = String.format("Successfully created property %s in %s",property, namespace);
+        return getResponseBuilder().buildResponse(response, "application/json", responseBody, HttpServletResponse.SC_CREATED); 
+    }
+
+    private void setProperty(String namespace, String propertyName, String value) throws FrameworkException {
+        if (!checkPropertyExists(namespace, propertyName)){
+            getFramework().getConfigurationPropertyService(namespace).setProperty(propertyName, value);
+        }else{
+            ServletError error = new ServletError(GAL5019_PROPERTY_ALREADY_EXISTS_ERROR, propertyName ,namespace);  
+            throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+        }
+    }
+
+    /*
+     * Handle Post Request
+     */
+    @Override
+    public HttpServletResponse handlePostRequest(String pathInfo, QueryParameters queryParameters,
+            HttpServletRequest request, HttpServletResponse response)
+            throws  IOException, FrameworkException {
+        String namespace = getNamespaceFromURL(pathInfo);
+        String property = getPropertyNameFromURL(pathInfo);
+        checkRequestHasContent(request, pathInfo);
+        String value = new String (request.getInputStream().readAllBytes(),StandardCharsets.UTF_8);
+        updateProperty(namespace, property, value);
+        String responseBody = String.format("Successfully updated property %s in %s",property, namespace);
+        return getResponseBuilder().buildResponse(response, "application/json", responseBody, HttpServletResponse.SC_CREATED); 
+    }
+
+    private void updateProperty(String namespace, String propertyName, String value) throws FrameworkException {
+        if (checkPropertyExists(namespace, propertyName)){
+            getFramework().getConfigurationPropertyService(namespace).setProperty(propertyName, value);
+        }else{
+            ServletError error = new ServletError(GAL5018_PROPERTY_DOES_NOT_EXIST_ERROR,propertyName);  
+            throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+        }
+    }
+
+    /*
+     * Handle Delete Request
+     */
+    public HttpServletResponse handleDeleteRequest(String pathInfo, QueryParameters queryParameters,
+            HttpServletRequest request, HttpServletResponse response)
+            throws FrameworkException {
+        String namespace = getNamespaceFromURL(pathInfo);
+        String property = getPropertyNameFromURL(pathInfo);
+        deleteProperty(namespace, property);
+        String responseBody = String.format("Successfully deleted property %s in %s",property, namespace);
+        return getResponseBuilder().buildResponse(response, "application/json", responseBody, HttpServletResponse.SC_OK);
+    }
+
+
+    private void deleteProperty(String namespace, String propertyName) throws FrameworkException {
+        if (checkPropertyExists(namespace, propertyName)){
```

Comment:
what if the namespace is hidden ? We don't want the REST interface users deleting dss properties I think.

Reply:
same question probably goes for all the GET/POST/PUT/DELETE request types.